apiserver/secretstorage/status/status.go

```diff
@@ -0,0 +1,20 @@
+package status
+
+import "sigs.k8s.io/apiserver-runtime/pkg/builder/resource"
+
+// ObjectWithStatusSubResource defines an interface for getting and setting the status sub-resource for a resource.
+// It is a copy of the interface in the apiserver-runtime package, but with the SecretStorage prefix.
+// The prefix is needed since the apiserver-runtime package fails if the storage is not a k8s.io/apiserver/pkg/registry/generic/registry.Store.
+type ObjectWithStatusSubResource interface {
+	resource.Object
+
+	// SecretStorageGetStatus should return the status sub-resource for the resource.
+	SecretStorageGetStatus() (statusSubResource StatusSubResource)
```

Comment:
This is a horrible hack i know 😅 But should allow for easily switching back to the official interfaces if upstream fixes storage registration.

Reply:
Took be a bit to understand this, but I guess this is fine